datatypes/src/collections/feature_collection_builder.rs

```diff
@@ -286,43 +295,131 @@ where
         self.len() == 0
     }
 
-    /// Outputs the number of bytes that is occupied in the builder buffers
+    /// Outputs the memory size in bytes that the feature collection will have after building
     ///
     /// # Panics
     /// * if the data type is unknown or unexpected which must not be the case
     ///
-    pub fn byte_size(&mut self) -> usize {
-        let geometry_size = CollectionType::builder_byte_size(&mut self.geometries_builder);
-        let time_intervals_size = TimeInterval::builder_byte_size(&mut self.time_intervals_builder);
+    pub fn estimate_memory_size(&mut self) -> usize {
+        // TODO: store information? avoid re-calculation?
+        let map_size = std::mem::size_of_val(&self.types)
+            + self
+                .types
+                .iter()
+                .map(|(k, v)| {
+                    std::mem::size_of_val(k) + k.as_bytes().len() + std::mem::size_of_val(v)
+                })
+                .sum::<usize>();
+
+        let geometry_size =
+            CollectionType::estimate_array_memory_size(&mut self.geometries_builder);
+
+        let time_intervals_size =
+            TimeInterval::estimate_array_memory_size(&mut self.time_intervals_builder);
 
         let attributes_size = self
             .builders
-            .values()
-            .map(|builder| {
+            .iter()
+            .map(|(name, builder)| {
                 let values_size = if builder.as_any().is::<Float64Builder>() {
-                    builder.len() * std::mem::size_of::<f64>()
+                    // Float
```

Comment:
this is a lot of duplicated code. Maybe you can use a generic function or macro for that.

Reply:
[5d17827](https://github.com/geo-engine/geoengine/pull/810/commits/5d17827f1309cd0ac07d6275a807f848e248fdea)